src/Microsoft.AspNetCore.Routing/Constraints/StringRouteConstraint.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing.Constraints
+{
+    /// <summary>
+    /// Constrains a route parameter to contain only a specifyed strign.
+    /// </summary>
+    public class StringRouteConstraint : IRouteConstraint
+    {
+        private string _value;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StringRouteConstraint"/> class.
+        /// </summary>
+        /// <param name="value">The constraint value to match.</param>
+        public StringRouteConstraint(string value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            _value = value;
+        }
+
+        /// <inheritdoc />
+        public bool Match(HttpContext httpContext, IRouter route, string routeKey, RouteValueDictionary values, RouteDirection routeDirection)
+        {
+            if (httpContext == null)
+            {
+                throw new ArgumentNullException(nameof(httpContext));
+            }
+
+            if (route == null)
+            {
+                throw new ArgumentNullException(nameof(route));
+            }
+
+            if (routeKey == null)
+            {
+                throw new ArgumentNullException(nameof(routeKey));
+            }
+
+            if (values == null)
+            {
+                throw new ArgumentNullException(nameof(values));
+            }
+
+            object routeValue;
+
+            if (values.TryGetValue(routeKey, out routeValue)
+                && routeValue != null)
+            {
+                var parameterValueString = Convert.ToString(routeValue, CultureInfo.InvariantCulture);
+
```

Comment:
I think it's best the way it is in the other constraints :+1: 

The one gotcha here is that `Convert.ToString((object)null)` will return empty string when provided with null. So it would appear (to someone debugging) that `routeValue` was provided as the empty string when in reality it wasn't present.